full-service/src/service/memo.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{
+    db::{
+        account::{AccountID, AccountModel},
+        models::Account,
+        WalletDbError,
+    },
+    service::ledger::{LedgerService, LedgerServiceError},
+    util::b58::{b58_decode_public_address, B58Error},
+    WalletService,
+};
+use mc_account_keys::AccountKey;
+use mc_connection::{BlockchainConnection, UserTxConnection};
+use mc_crypto_keys::{KeyError, RistrettoPublic};
+use mc_fog_report_validation::FogPubkeyResolver;
+use mc_transaction_extra::{MemoDecodingError, MemoType, UnusedMemo};
+use mc_util_serial::DecodeError;
+use std::{convert::TryFrom, ops::DerefMut};
+use thiserror::Error;
```

Comment:
Why this and not `displaydoc` like we've used everywhere else?

Reply:
starting to move over to thiserror. It will reduce the cumbersome and boilerplate impl From<ErrorType> for <ErrorType>